src/file/FileUtils.js

```diff
@@ -434,19 +434,24 @@ define(function (require, exports, module) {
      * @return {number} The result of the local compare function
      */
     function compareFilenames(filename1, filename2, extFirst) {
-        var ext1   = getFileExtension(filename1),
-            ext2   = getFileExtension(filename2),
-            lang   = brackets.getLocale(),
-            cmpExt = ext1.toLocaleLowerCase().localeCompare(ext2.toLocaleLowerCase(), lang, {numeric: true}),
-            cmpNames;
+        var lang = brackets.getLocale();
         
-        if (brackets.platform === "win") {
-            filename1 = getFilenameWithoutExtension(filename1);
-            filename2 = getFilenameWithoutExtension(filename2);
+        function cmpExt() {
+            var ext1 = getFileExtension(filename1),
+                ext2 = getFileExtension(filename2);
+            return ext1.toLocaleLowerCase().localeCompare(ext2.toLocaleLowerCase(), lang, {numeric: true});
+        }
+        
+        function cmpNames() {
+            if (brackets.platform === "win") {
```

Comment:
If you're interested in another way to solve the file system sorting problem, take a look at the `getSortedFiles()` function I wrote in the original PR [here](https://github.com/adobe/brackets/pull/11128/files#diff-0ba586b1d4dc4a89d8405edd63a72644R371).

To sort fullpaths, `comparePaths()` splits each path into an array of folders with last item being filename. This is a good technique, but it *splits both paths for every, single fullpath comparison*. In `getSortFiles()`, it splits paths all of the file before sorting starts (so it only happens once for each file) and then executes the sort. This seems to reduce sorting time by another ~16%.

I did not suggest suggest this to be taken for Brackets because it only handles the sort by filename case, not file extension required by the working set, and it's currently not called anywhere in master, so it would need to be factored in somewhere.